local.tf

```diff
@@ -0,0 +1,566 @@
+    Name = "wandb-eks-master"
+  }
+}
+
+resource "aws_eks_cluster" "wandb" {
+  name     = "wandb"
+  role_arn = aws_iam_role.wandb_cluster_role.arn
+  version  = "1.15"
+
+  vpc_config {
+    security_group_ids = [aws_security_group.eks_master.id]
+    subnet_ids         = aws_subnet.wandb.*.id
+  }
+
+  depends_on = [
+    aws_iam_role_policy_attachment.wandb_eks_cluster_policy,
+    aws_iam_role_policy_attachment.wandb_eks_service_policy,
+  ]
+}
+
+resource "aws_security_group_rule" "eks_worker_ingress" {
+  description              = "Allow comntainer NodePort service to receive load balancer traffic"
+  protocol                 = "tcp"
+  security_group_id        = aws_eks_cluster.wandb.vpc_config[0].cluster_security_group_id
+  source_security_group_id = aws_security_group.wandb_alb.id
+  from_port                = 32543
+  to_port                  = 32543
+  type                     = "ingress"
+}
+
+resource "aws_iam_role" "wandb_cluster_role" {
+  name = "wandb-cluster-role"
+
+  assume_role_policy = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "eks.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+POLICY
+}
+
+resource "aws_iam_role_policy_attachment" "wandb_eks_cluster_policy" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy"
+  role       = aws_iam_role.wandb_cluster_role.name
+}
+
+resource "aws_iam_role_policy_attachment" "wandb_eks_service_policy" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEKSServicePolicy"
+  role       = aws_iam_role.wandb_cluster_role.name
+}
+
+data "aws_iam_policy_document" "wandb_node_policy" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "wandb_node_role" {
+  name               = "wandb-eks-node"
+  assume_role_policy = data.aws_iam_policy_document.wandb_node_policy.json
+}
+
+resource "aws_iam_role_policy_attachment" "wandb_node_worker_policy" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy"
+  role       = aws_iam_role.wandb_node_role.name
+}
+
+resource "aws_iam_role_policy_attachment" "wandb_node_cni_policy" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy"
+  role       = aws_iam_role.wandb_node_role.name
+}
+
+resource "aws_iam_role_policy_attachment" "wandb_node_registry_policy" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly"
+  role       = aws_iam_role.wandb_node_role.name
+}
+
+resource "aws_iam_role_policy_attachment" "wandb_node_s3_policy" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonS3FullAccess"
+  role       = aws_iam_role.wandb_node_role.name
+}
+
+resource "aws_iam_role_policy_attachment" "wandb_node_sqs_policy" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonSQSFullAccess"
+  role       = aws_iam_role.wandb_node_role.name
+}
```

Comment:
If customers end up using these roles in their existing accounts, they would want the policies to only grant access to the bucket and queues we create.  Even if we're doing the sub account it's best practice to grant least privilege...

Reply:
Good point, I'll write a custom policy on the next pass.

Reply:
Fixed now.